larray/core.py

```diff
@@ -4312,13 +6153,36 @@ def transpose(self, *args):
     T = property(transpose)
 
     def clip(self, a_min, a_max, out=None):
+        """Clip (limit) the values in an array.
+
+        Given an interval, values outside the interval are clipped to the interval edges.
+        For example, if an interval of [0, 1] is specified, values smaller than 0 become 0,
+        and values larger than 1 become 1.
+
+        Parameters:
+        -----------
+        a_min : scalar or array-like
+            Minimum value.
+        a_max : scalar or array-like
+            Maximum value.  If `a_min` or `a_max` are array_like, then they will
```

Comment:
it goes beyond that, as all 3 arguments (self, amin and amax) will be broadcasted to the axes resulting from combining all 3 arrays. in other words, amin & amax can have more axes than self, which is actually useful. say you want to apply a limit which is different by sex. The result will have a sex axis even if the original array did not.